adoc-mode.el

```diff
@@ -301,6 +308,27 @@ Also used to delimit the scan for the end delimiter."
   :group 'adoc
   :package-version '(adoc-mode . "0.8.0"))
 
+(defcustom adoc-max-image-size nil
+  "Maximum width and height for displayed images.
+This variable may be nil or a cons cell (MAX-WIDTH . MAX-HEIGHT).
+When nil, use the actual size.  Otherwise, use ImageMagick to
+resize larger images to be of the given maximum dimensions.  This
+requires Emacs to be built with ImageMagick support."
+  :group 'adoc
+  :package-version '(adoc-mode . "0.9.0")
+  :type '(choice
+          (const :tag "Use actual image width" nil)
+          (cons (choice (sexp :tag "Maximum width in pixels")
+                        (const :tag "No maximum width" nil))
+                (choice (sexp :tag "Maximum height in pixels")
+                        (const :tag "No maximum height" nil)))))
+
+(defcustom adoc-display-images-at-startup t
```

Comment:
In general the display state of the images are solely controlled by the image overlays in the buffer. So there is no variable needed for that.

The sense of `adoc-display-images-at-startup` is to give the user the possibility to avoid the preview of images when an adoc-file is opened.

You are right, that `at-startup` is a bad suffix, since the "Emacs startup sequence" is a reserved term. Maybe it is better to call it `adoc-display-images-at-file-open`?

Reply:
@bbatsov I've got the feeling that I miss the point of what you want with `adoc-display-images`. I intended it as a user option. But, I have the feeling that you actually want it to be a buffer-local variable.

The user has the possibility to remove the preview for a single image link with `adoc-remove-image-overlay-at`. I know from my writing experience with Orgmode that this is an important feature.
Furthermore, the user can insert a new image link having no overlay yet.
Both actions lead to a state where some image links in the buffer have overlays and some have none. This is not a clear state that could be described easily by a buffer-local variable `adoc-display-images`.
This is also the reason for `adoc-toggle-images` for first removing all remaining overlays and returning t if overlays were removed. If no overlays were removed it re-generates all overlays.
So, if there is a mixed state `adoc-toggle-images` first establishes the clean state that there are no overlays at all. And if this state is reached the next call of `adoc-toggle-images` will regenerate all overlays.

Reply:
Yeah, I was thinking of making it `buffer-local`, which is not in direct conflict with it being an user option. 

>  Both actions lead to a state where some image links in the buffer have overlays and some have none. This is not a clear state that could be described easily by a buffer-local variable adoc-display-images.

You can always reset everything at toggle time. My point is mostly that this variable in its current form seems too specific to be of much usefulness. If you're concerned about changing it, I'd rather just remove it or leave it as-is, but with the name that I suggested, so we have more flexibility to change its behavior down the road.

Reply:
>  This is also the reason for adoc-toggle-images for first removing all remaining overlays and returning t if overlays were removed. If no overlays were removed it re-generates all overlays.

Btw, it's good to also mention such more granular use-cases in the README, as I'm normally an "all-in" user, so the thought of showing/hiding particular images didn't even cross my mind and I'm pretty sure I wouldn't be the only person thinking like this.